src/light_data.c

```diff
@@ -661,9 +826,38 @@ void update_light_render_area(void)
     light_render_area(startx, starty, endx, endy);
 }
 
-void light_set_light_minimum_size_to_cache(long a1, long a2, long a3)
+void light_set_light_minimum_size_to_cache(long lgt_id, long a2, long a3)
 {
-  _DK_light_set_light_minimum_size_to_cache(a1, a2, a3);
+  // _DK_light_set_light_minimum_size_to_cache(a1, a2, a3);
+  struct Light *lgt;
+  if ( lgt_id )
+  {
+    lgt = &game.lish.lights[lgt_id];
+    if ( lgt->flags & LgtF_Allocated )
+    {
+      if ( lgt->flags & LgtF_Unkn02 )
+      {
+        lgt->flags &= ~2;
+        if ( lgt->flags & LgtF_Dynamic )
+        {
+          lgt->field_9[0] = a2;
```

Comment:
Could you please break `field_9` (as array of size 5) into `field_9` and `field_A[4]`.
Also I am unsure is `field_9` signed or not (a2 is signed).
Also a2 is either byte or signed char.